### DISCUSSION TITLE
Improving Server Workshop Information

### DISCUSSION THREAD

**Discussion by Gamingtoday093** [SmartlyDressedGames/Unturned-3.x-Community#4114] [Ideas]
The current server workshop content preview section in the server browser does not contain enough information about the mods the server uses. Mainly how much space they will take.

Each workshop item should have their size displayed as well as a total amount for all of the items combined. This would make browsing for servers that don't have several gigabytes of mods a lot easier. You could even include a warning label for servers that exceed some absurd limit like 5 GB.

Another related issue is that all mods are treated equally when they really shouldn't be. Right now a 300 Gun 100 mb mod takes up and displays the exact same way as a tiny plugin UI mod under 1 mb. Servers currently solve this issue by taking all of the small plugin-based workshop mods and packaging them into one larger mod. This is simply cumbersome and annoying for everyone involved. For server owners having to package and update many mods, for plugin developers not being able to display their cool workshop UI mod in the server description and for players trying to view what mods a server is running.

It'd be a lot better if workshop items were sorted by size. Not in the listed items but rather splitting up into smaller and bigger mods. Bigger mods could remain as they are now. But smaller mods (Something like less than 2mb) should be relegated to an expandable list dropdown-like menu which would better describe the server's workshop content.

Here is a mock-up of what I had in mind:
![Improved Workshop Page](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/70017362/c0de49a4-41fc-4c3b-9a38-f7586d71f5b8)

**Comment by KarmaWSYD** [comment]
Since workshop mods are (at least as far as I understand) loaded in RAM if this were to be implemented it'd also be nice to have a warning for when the amount exceeds some arbitrary figure (Say, 4GB + workshop mod count) compared to the system memory, given that system memory is already being detected by the game. Some people with older and lower-end machines can sometimes have issues with even joining servers with loads of mods as their computer runs out of memory.

Just having the total workshop file size by itself would be useful to have though, given that it could help with estimating performance (esp. on the mentioned low-end computers) and how long it takes to load into the server.

**Comment by Jdance-Media** [comment]
TalkinTurtle recently requested about a workshop filter on the server list that sorts by mod count. What Gaming is presenting here is related and would be somewhat helpful to players. Alongside that, perhaps Unturned should be monitoring system memory during the asset loading process. If the system memory is nearing full utilization, then the game should warn the player and ask if they want to continue.

**Comment by yzdjmm-unturned** [comment]
I completely agree, I don't like servers with more than 8 mods on the Workshop interface, because I am not only worried about whether my memory capacity will be taken up a lot, but also because I don't want to play this server, and when I need to unsubscribe from the Workshop mods in the future, I have to search for them in the Workshop interface and unsubscribe one by one, which will waste a lot of time

**Reply by GazziFX** [reply]
You don't need to manually subscribe/unsubscribe when joining the server, but mods will still persist on disk that can be annoying. I think the auto clear can be implemented if asset not used for N amount of time
